dev_tools/.githooks/setup_git_hooks.sh

```
@@ -0,0 +1,47 @@
+# Copy all hook scripts to the Git hooks directory
+echo "🚀 Copying hook scripts..."
+if ! cp -r "$SCRIPT_DIR"/* "$GIT_HOOKS_DIR"; then
+    echo "❌ Error: Failed to copy hook scripts."
+    exit 1
+fi
+
+# Identify newly copied files (files that were NOT present before)
+declare -a NEW_FILES=()
+for file in "$GIT_HOOKS_DIR"/*; do
+    filename=$(basename "$file")
+    if [[ ! " $PRE_EXISTING_FILES " =~ " $filename " ]]; then
```

Shellcheck (0.10.0)

[warning] Remove quotes from right-hand side of =~ to match as a regex rather than literally.

(SC2076)